test/lib/jdk/test/lib/hotspot/ir_framework/TestFrameworkExecution.java

```diff
@@ -0,0 +1,1452 @@
+                    try {
+                        applyIndependentCompilationCommands(ex);
+                    } catch (TestFormatException e) {
+                        // Failure logged. Continue and report later.
+                    }
+                }
+
+                // Only force compilation now because above annotations affect inlining
+                for (Executable ex : executables) {
+                    try {
+                        applyForceCompileCommand(ex);
+                    } catch (TestFormatException e) {
+                        // Failure logged. Continue and report later.
+                    }
+                }
+            }
+        }
+    }
+
+    private void applyClassAnnotations(Class<?> c) {
+        ForceCompileClassInitializer anno = getAnnotation(c, ForceCompileClassInitializer.class);
+        if (anno != null) {
+            // Compile class initializer
+            CompLevel level = anno.value();
+            if (level == CompLevel.SKIP || level == CompLevel.WAIT_FOR_COMPILATION) {
+                TestFormat.failNoThrow("Cannot define compilation level SKIP or WAIT_FOR_COMPILATION in " +
+                                       "@ForceCompileClassInitializer at " + c);
+                return;
+            }
+            level = restrictCompLevel(anno.value());
+            if (level != CompLevel.SKIP) {
+                // Make sure class is initialized to avoid compilation bailout of <clinit>
+                getClassObject(c.getName(), "nested"); // calls Class.forName() to initialize 'c'
+                TestFormat.checkNoThrow(WHITE_BOX.enqueueInitializerForCompilation(c, level.getValue()),
+                                        "Failed to enqueue <clinit> of " + c + " for compilation. Did you specify "
+                                        + "@ForceCompileClassInitializer without providing a static class initialization? "
+                                        + "Make sure to provide any form of static initialization or remove the annotation.");
+            }
+        }
+    }
+
+    private void checkClassAnnotations(Executable ex) {
+        TestFormat.checkNoThrow(getAnnotation(ex, ForceCompileClassInitializer.class) == null,
+                                "@ForceCompileClassInitializer only allowed at classes but not at method " + ex);
+    }
+
+    /**
+     * Exclude a method from compilation randomly and returns the compilation level on which a compilation is still
+     * possible.
+     */
+    static CompLevel excludeRandomly(Executable ex) {
+        Random random = Utils.getRandomInstance();
+        boolean exclude = random.nextBoolean();
+        CompLevel level = CompLevel.ANY;
+        if (exclude) {
+            String levelName;
+            switch (random.nextInt() % 3) {
+                case 1 -> {
+                    level = CompLevel.C1;
+                    WHITE_BOX.makeMethodNotCompilable(ex, CompLevel.C2.getValue(), false);
```

Comment:
I'm curious why two calls (with different values for the last parameter) are needed for these?

Reply:
The parameter is for [osr](https://github.com/openjdk/jdk/blob/dc323a933401e4e540cdc416de5260923f42ba7f/src/hotspot/share/prims/whitebox.cpp#L928-L937): We need to exclude the method for osr and for normal compilations.